### PR TITLE
ENSCORESW-2933: new CCDS parser with test case

### PR DESCRIPTION
**Pull request by magaliruffier**
## Requirements

- Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion;
- Review the [contributing guidelines](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#why-could-my-pull-request-be-rejected) for this repository; remember in particular:
    - do not modify code without testing for regression
    - provide simple unit tests to test the changes
    - if you change the schema you must patch the test databases as well, see [Updating the schema](https://github.com/Ensembl/ensembl/blob/master/CONTRIBUTING.md#updating-the-schema)
    - the PR must not fail unit testing

## Description

_Using one or more sentences, describe in detail the proposed changes._
Fresh implementation of CCDS parsing

## Use case

_Describe the problem. Please provide an example representing the motivation behind the need for having these changes in place._
Retrieves CCDS data from the core database and adds it as direct xref into the xref database
This replaces the functionality from the CCDSParser in the xref_mapping directory. Rather than requiring a dedicated CCDS database, the data is retrieved from the core database directly. It is Perl critic compliant and matches the standards as seen in other parsers in the repository

## Benefits

_If applicable, describe the advantages the changes will have._
CCDS data is added to the xref pipeline

## Possible Drawbacks

_If applicable, describe any possible undesirable consequence of the changes._

## Testing

_Have you added/modified unit tests to test the changes?_
Yes

_If so, do the tests pass/fail?_
Tests pass

_Have you run the entire test suite and no regression was detected?_
Yes

### CODE REVIEW

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+-species = only used by human and mouse
+
+=head1 SYNOPSIS
+
+  my $parser = Bio::EnsEMBL::Xref::Parser::CCDSParser->new(
+    source_id  => 1,
+    species_id => 9606,
+    dba        => $dba   # core db adaptor
+  );
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: xref_dba argument missing. Guess it needs it as well

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    SELECT ta.value, t.stable_id
+    FROM transcript t
+    INNER JOIN transcript_attrib ta ON t.transcript_id = ta.transcript_id
+    INNER JOIN attrib_type a ON ta.attrib_type_id = a.attrib_type_id
+    WHERE a.code = 'ccds_transcript';
+CCDS
+
+  my $xref_count = 0;
+  my $sth = $dba->dbc->prepare($sql);
+  $sth->execute() or croak( $dba->errstr() );
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: confess?!

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      species_id => $species_id,
+      info_type  => 'DIRECT'
+    } );
+    $xref_count++;
+  }
+  $sth->finish;
+
+  print "Added $xref_count DIRECT xrefs\n" if ($verbose);
+  if ( !$xref_count ) {
+   croak "No CCDS xref added\n";
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: confess!

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm (hunk 1 above)
Comment (magaliruffier) [reply modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: It is not used in this parser, but it is a required parameter, I have added it in the example

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm (hunk 2 above)
Comment (magaliruffier) [reply modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: sorry, updated now

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm (hunk 3 above)
Comment (magaliruffier) [reply modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: idem

File: modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+
+package Bio::EnsEMBL::Xref::Parser::CCDSParser;
+
+use strict;
+use warnings;
+use Carp;
+
+use parent qw( Bio::EnsEMBL::Xref::Parser );
+
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/CCDSParser.pm]: Perl critic fails here..
#   Failed test 'Pod coverage on Bio::EnsEMBL::Xref::Parser::CCDSParser'
#   at modules/t/pod_coverage.t line 43.
# Coverage for Bio::EnsEMBL::Xref::Parser::CCDSParser is 0.0%, with 1 naked subroutine:
# 	run

**Comment by premanand17** [comment]
Travis fails for other reasons unrelated to this PR.

**Pull request merged**